### ISSUE TITLE
`rumdl check --list-rules` is shown in help and README but ignored

### ISSUE THREAD

**Issue by ph0ryn** [rvben/rumdl#680]
### rumdl version

0.2.19

### Installation method

Other

### Bug description

`rumdl check -l` / `rumdl check --list-rules` is shown in `rumdl check --help`, but the flag appears to be ignored. Instead of listing rules, it runs a normal check.

I am not sure whether the intended fix is to keep this as a compatibility alias for `rumdl rule`, or to remove the flag from `rumdl check` now that `rumdl rule` exists.

### Steps to reproduce

```bash
# Empty directory case
tmp=$(mktemp -d)
cd "$tmp"
rumdl check -l
```

```bash
# File case
tmp=$(mktemp -d)
printf '#Title\n' > "$tmp/bad.md"
cd "$tmp"
rumdl check -l bad.md
```

I also checked current `main` (`b7d94ba8`, v0.2.19) with:

```bash
cargo run --quiet -- check -l
```

### Expected behavior

Either:

1. `rumdl check -l` / `rumdl check --list-rules` should list rules, matching the help text; or
2. `-l, --list-rules` should be removed from `rumdl check`, since `rumdl rule` is the current rule-listing command.

I wanted to ask which direction you prefer before opening a PR.

### Actual behavior

In an empty directory, `rumdl check -l` runs the normal check path and reports:

```text
No markdown files found to check.
```

With a file, it lints the file instead of listing rules:

```text
bad.md:1:2: [MD018] No space after # in heading [*]

Issues: Found 1 issues in 1 file
```

On current `main`, `cargo run --quiet -- check -l` also runs a normal check instead of listing rules.

### Configuration

```toml
No configuration required to reproduce.
```

### Operating system

macOS

### Additional context

I noticed `CheckArgs` still defines `list_rules`, but I could not find any use of `args.list_rules` in the check command path.

There is a related older issue, #73, which suggests `rumdl rule` replaced `--list-rules`, so removing the inert `check --list-rules` flag may be preferable to implementing it as an alias.

For context:
I was looking to migrate from `markdownlint-cli2` and just found rumdl today. 
At first I ran `rumdl check -l` thinking it might show which rules were active, so I could confirm that my existing config was being loaded. It printed nothing useful for that, so I checked the README and realized the option is described as “List all available rules”.
That made me notice this issue.
I’m currently learning English and used Codex to help write this, but I did confirm the issue reproduces on my local machine before opening this.

**Comment by rvben** [comment]
Thanks, @ph0ryn, great catch, and thanks for offering a PR. Resolved in v0.2.20.

You were right that the flag was inert. We removed it from `check`/`fmt` rather than aliasing it, since rule listing already has dedicated commands. To avoid the confusing dead-end you hit, `--list-rules` / `-l` is now hidden and, if used, prints guidance instead of a bare error:

```
Error: `--list-rules` has been removed
  To list all rules:                   rumdl rule
  Rules active for your config:        rumdl check --verbose
  Inspect your effective config:       rumdl config
```

For your original goal (confirming the config loaded / which rules are active), `rumdl config` shows the effective config with provenance and `rumdl check --verbose` lists the active rules. The README no longer advertises the flag. Closing, and thanks for the thorough write-up.

**Comment by ph0ryn** [comment]
Thanks for the quick fix and for clarifying the intended commands. It looks helpful!
